src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -115,13 +111,25 @@ private static bool IsThrottled(OAuth2ResponseBase oAuth2Response)
         {
             MsalServiceException ex = new MsalServiceException(errorCode, errorMessage, innerException);
 
-            ex.ResponseBody = httpResponse?.Body;
-            ex.StatusCode = httpResponse != null ? (int)httpResponse.StatusCode : 0;
-            ex.Headers = httpResponse?.Headers;
+            SetHttpExceptionData(ex, httpResponse);
 
             return ex;
         }
 
+        internal static MsalThrottledServiceException FromThrottledCLientCredentialResponse(HttpResponse httpResponse)

```

Comment:
Throttling can occur on any flow, not just client credentials. The method name should reflect that.